api/types/src/convert.rs

```diff
@@ -656,40 +629,8 @@ impl<'a, S: StateView> MoveConverter<'a, S> {
 
         let ret = match payload {
             TransactionPayload::EntryFunctionPayload(entry_func_payload) => {
-                let EntryFunctionPayload {
-                    function,
-                    type_arguments,
-                    arguments,
-                } = entry_func_payload;
-
-                let module = function.module.clone();
-                let code =
-                    self.inner.view_existing_module(&module.clone().into())? as Arc<dyn Bytecode>;
-                let func = code
-                    .find_entry_function(function.name.0.as_ident_str())
-                    .ok_or_else(|| format_err!("could not find entry function by {}", function))?;
-                ensure!(
-                    func.generic_type_params.len() == type_arguments.len(),
-                    "expect {} type arguments for entry function {}, but got {}",
-                    func.generic_type_params.len(),
-                    function,
-                    type_arguments.len()
-                );
-                let args = self
-                    .try_into_vm_values(func, arguments)?
-                    .iter()
-                    .map(bcs::to_bytes)
-                    .collect::<Result<_, bcs::Error>>()?;
-
-                Target::EntryFunction(EntryFunction::new(
-                    module.into(),
-                    function.name.into(),
-                    type_arguments
-                        .into_iter()
-                        .map(|v| v.try_into())
-                        .collect::<Result<_>>()?,
-                    args,
-                ))
+                let entry_function = self.try_to_aptos_core_entry_function(entry_func_payload)?;
```

Comment:
Same suggestion, if this is for type coercion, perhaps we should use `into`. If this is a function define by us, perhaps we may use `supra_core` instead?